First draft of precourse docs

jsms90 opened the pull request:
Learning outcomes - from F&C9 gitbook + new topics
- review & changes welcome

Resources - heavily altered
- give reasons for not using frameworks
- command line tutorial from playbook
- git & github: part of Udacity, remove the rest
- add markdown tutorial from playbook
- add semantic HTML to help with week 1
- add resources to help with common CSS problems
- remove callback hell (too complex), add event loop video
- remove Eloquent Javascript (too long)
- reduce number of DOM manipulation resources
- reorder so that resources lead up to a project (which is only "recommended", where rest of topics are "expected"). Use advanced instead of intermediate FCC project to avoid API requests

Installation list - from F&C9 gitbook

Related #26 - could close this issue on merging, to avoid F&C10 seeing other resources that were debated and feeling any pressure?

jsms90 commented:
I've just realised that we're only recommending the first 4 lessons in Zapier's API tutorial. @njsfield Do you remember why?

bradreeder commented:
Will hold off until review is complete @jsms90 @Cleop @njsfield

njsfield commented on coursebook/precourse/installation-list.md:
```diff
@@ -0,0 +1,37 @@
+- Install Chrome
+- [Setup SSH access to your Github account](https://help.github.com/articles/generating-an-ssh-key/)
+- Atom plugins:
+  - linter & linter-eslint
```
Ah, I just realised these plugins don't work unless the eslint modules are installed (and linting is covered later in Node Week 1). To avoid confusion it may be worth removing these plugins from the install guide

bradreeder replied:
Do you need the atom plugins installed in order to use eslint with atom? Would be worth keeping it in that case to save people time in node week.

njsfield commented on coursebook/precourse/resources.md:
```diff
@@ -0,0 +1,67 @@
+
+## Command Line
+
+Use this [Command Line Interface Crash Course](https://learnpythonthehardway.org/book/appendixa.html) to get familiar with navigating around your computer without using a GUI.
```
Perhaps add a small comment that this resource is taken from an unrelated python course (to prevent confusion)

njsfield commented on coursebook/precourse/resources.md:
```diff
@@ -0,0 +1,67 @@
+
+Use this [Command Line Interface Crash Course](https://learnpythonthehardway.org/book/appendixa.html) to get familiar with navigating around your computer without using a GUI.
+
+## Git & Github
```
to avoid confusion perhaps clarify 
>lesson 1 (Navigating A Commit History section)
as lesson 1 may be interpreted as just the introduction video (highly unlikely but just incase)

jsms90 commented:
@bradreeder feel free to review too. we have added and removed a fair bit. would be good to get your input on anything, with the benefit of slightly more experience than us

njsfield commented on coursebook/precourse/resources.md:
```diff
@@ -0,0 +1,67 @@
+
+## APIs
+
+[Learn APIS: First 4 chapters](https://zapier.com/learn/apis/) introduces basics of APIS, HTTP, and API keys
```
It may be worth only suggesting only the first 3 chapters and updating the description to 
> Learn APIS: First 4 chapters introduces the basics of HTTP and APIs

as API keys are a lot deeper than the broader topics which are introduced here

njsfield commented on coursebook/precourse/resources.md:
```diff
@@ -0,0 +1,67 @@
+
+## Practical Project
+
+The main [objective](learning-objectives.md) of your precourse material is to become comfortable with the topics in the material above. However, from our experience, you'll find working on a personal project _very_ useful to cement this knowledge. We recommend that you try putting this all into practice by building a calculator, as suggested in the [FCC Advanced Front End Development Projects](ht ...
```
Maybe change to
> We recommend that you build a calculator, and apply some of what you've learnt in these resources.
As some content like API/Http theory/callbacks may not be applicable to the project

bradreeder commented on coursebook/precourse/resources.md:
```diff
@@ -0,0 +1,67 @@
+
+If you have any problems with the following material, feel free to reach out to us through your cohort's gitter channel!
+
+## Command Line
```
Did you all discuss potentially providing a priority order for these resources? I'd prefer everyone started with the things we won't get a chance to teach them, like learning to use the CLI and writing markdown, than things we will as that saves us time if they arrive unprepared? It might give them a sense of an order to progress through the material.

bradreeder replied:
Don't forget that many of you felt intimidated by the precourse. If people have at least done the highest priority tasks they might feel a bit better.

bradreeder commented on coursebook/precourse/resources.md:
```diff
@@ -0,0 +1,67 @@
+
+F&C is a very intense course, and you should make sure you're well-rested and ready to work when you get here. Below is a list of material that will help you to be prepared for the start of the course. The list might look a bit long, but each resource itself is quite short. Doing bitesized chunks little and often should make this manageable before the course begins. Your curriculum is designed wi ...
+
+Please do not forget that **we don't use frameworks at FAC** – ignore Bootstrap and JQuery, and make sure you work in vanilla HTML, CSS and JavaScript. We want you to build your skills without being reliant on any one framework so that you fully appreciate what problems each framework is solving, and are able to therefore make more effective judgements on which technologies to use in future. Anot ...
```
Your reasoning is sound and is more informative -- and this is only my preference -- but I have a feeling the paragraph is too long and should be quick to read.

"Please do not forget that **we don't use frameworks at FAC** -- ignore Bootstrap and JQuery. We believe that if you work in vanilla HTML, CSS and Javascript first you will gain a deeper understanding of the fundamentals that frameworks and libraries build upon and the problems that they solve for you."

jsms90 replied:
Yeah. You're right. It's a wall of text, and is going to be annoying when they keep returning to the page while they're completing everything.

How about we just remove the explanation from here altogether and put a link to general-->house rules, and put this fuller explanation into that file? I like the idea of fleshing out our reasoning, like @nelsonic 's love for starting with the 'why'.

bradreeder commented on coursebook/precourse/resources.md:
```diff
@@ -0,0 +1,67 @@
+
+Congratulations to all of you on being part of the next F&C cohort!
+
+F&C is a very intense course, and you should make sure you're well-rested and ready to work when you get here. Below is a list of material that will help you to be prepared for the start of the course. The list might look a bit long, but each resource itself is quite short. Doing bitesized chunks little and often should make this manageable before the course begins. Your curriculum is designed wi ...
```
Again, only preference, but maybe soften a little. Others feel free to weigh in.

* "F&C is an intense course" [very intense _eek_]
* "Below is a list of material that will help you prepare for the start of the course" [being 'prepared' implies if you don't do the precourse you won't be prepared]
* "Your curriculum is designed with the assumption that you'll have some familiarity with these topics by week one."

jsms90 replied:
I agree with all but the second point. That implication is exactly why I want this line in here. We wanted to move towards a non-optional precourse. I don't think the "optional" nature helped my cohort, especially when there was teaching that assumed we had read/watched what was suggested in our precourse.

bradreeder commented on coursebook/precourse/resources.md:
```diff
@@ -0,0 +1,67 @@
+# Resources
+
+Congratulations to all of you on being part of the next F&C cohort!
```
Maybe irrelevant if the online version of the curriculum became a separate folder.

Cleop commented on coursebook/precourse/learning-outcomes.md:
```diff
@@ -0,0 +1,50 @@
+
+### Semantic HTML
+
+- Use new HTML5 tags that will improve symantics
```
semantics*

bradreeder commented on coursebook/general/house-rules.md:
```diff
@@ -10,4 +10,7 @@
 
-+ **No frameworks**. For the first 10 weeks of the course, stick to vanilla JavaScript. That means no frameworks, backend or frontend (i.e. no Bootstrap/jQuery etc...). This is for the benefit of your understanding of JavaScript; trust us (and all the previous FAC cohorts) – it works!
++ **No frameworks**. For the first 10 weeks of the course, stick to vanilla JavaScript. That means no frontend frameworks (i.e. no Bootstrap/jQuery etc...). We want you to build your skills without being reliant on these.
+  + If you fully appreciate what problems each framework is solving, you are then able to make more effective judgements on which technologies to use in future.
```
"By solving problems using the foundations of the language first, you will begin to understand what problems frameworks were built to solve. This will let you make more effective judgements on which technologies to use in future."

Unsure if current phrasing makes it clear why you will appreciate what problems frameworks solve by using vanilla js. Can we assume that everyone knows that under the surface jQuery is written in vanilla javascript?

bradreeder commented on coursebook/general/house-rules.md:
```diff
@@ -10,4 +10,7 @@
 + **20-minute rule**. Struggle is good, but not too much of it. As a general rule, if you cannot solve a problem within 20 minutes, then stop and talk to somebody about it (there should be a timer on your desk)
 
-+ **No frameworks**. For the first 10 weeks of the course, stick to vanilla JavaScript. That means no frameworks, backend or frontend (i.e. no Bootstrap/jQuery etc...). This is for the benefit of your understanding of JavaScript; trust us (and all the previous FAC cohorts) – it works!
++ **No frameworks**. For the first 10 weeks of the course, stick to vanilla JavaScript. That means no frontend frameworks (i.e. no Bootstrap/jQuery etc...). We want you to build your skills without being reliant on these.
```
"stick to vanilla HTML, CSS, and JavaScript"?
